### PR TITLE
RND-2474 --- BEIR improve and evaluate

### PR DESCRIPTION
**Pull request by kanagnostopoulos**
In this PR I have:
- Fix requirements installation.
- Placed wrongly placed code inside /scripts or /beir folders.
- Redesigned the whole experiment pipeline to log all metrics in the same run so that everything is placed inside an experiment called "beir".

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,13 @@
+elasticsearch==7.9.1
+faiss_cpu
```
Comment (stevaras2) [inline requirements.in]: probably we need to define the version of this library

Comment (kanagnostopoulos) [reply requirements.in]: True, there were multiple libraries without a version specified. Should I place there the latest ones that were used?

File: scripts/run_dense_experiment_instruction.py
Comment (stevaras2) [inline scripts/run_dense_experiment_instruction.py:1]: Probably you can merge this one with the previous one script. The "instructions" parameter can be there for each occasion. For the dense experiment without "instruction", this parameter can be empty.

**Review by stevaras2 — APPROVED** [review]

File: scripts/run_dense_experiment_instruction.py
Comment (kanagnostopoulos) [reply scripts/run_dense_experiment_instruction.py:1]: And still use the OnnxBGE, even with an empty instruction?

Comment (stevaras2) [reply scripts/run_dense_experiment_instruction.py:1]: yes the results can be exactly the same

Comment (kanagnostopoulos) [reply scripts/run_dense_experiment_instruction.py:1]: OK, I will proceed with the changes.

Comment (kanagnostopoulos) [reply scripts/run_dense_experiment_instruction.py:1]: Done

File: requirements.in (hunk 1 above)
Comment (kanagnostopoulos) [reply requirements.in]: Done

**Review by CoGian — APPROVED** [review]
In the future, we should update beir to its newest version and support python 3.10.

**Comment by kanagnostopoulos** [comment]
Furthermore, in the future, all the other experiments (reranking, crossencoder etc) must be also edited to properly use the same philosophy for experiment tracking. I could raise an issue for this but I can't since it is a forked repo.

**Pull request merged**